components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from "react";
+import styled, { FlattenSimpleInterpolation, css } from "styled-components";
+
+interface Props extends React.Attributes {
+  multiline?: boolean;
+  rows?: number;
+  placeholder?: string;
+  size?: TextInputSize;
+  onChange?: (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => void;
+  className?: string;
+  style?: React.CSSProperties;
+}
+
+export default function InputText({
+  multiline = false,
+  placeholder = "",
+  rows = 0,
+  size = TextInputSize.regular,
+  onChange,
+  ...props
+}: Props) {
+  return multiline ? (
+    <TextAreaRoot
+      rows={rows}
+      placeholder={placeholder}
+      size={size}
+      onChange={onChange}
+      {...props}
+    />
+  ) : (
+    <TextInputRoot
+      type="text"
+      placeholder={placeholder}
+      size={size}
+      onChange={onChange}
+      {...props}
+    />
+  );
+}
+
+export enum TextInputSize {
+  fullWidth,
+  large,
+  regular,
+  small
+}
+
+const TextInputRoot = styled.input<{
+  size: TextInputSize;
+}>`
+  background-color: #fff;
+  border: solid 1px #576574;
+  border-radius: 4px;
+  color: #576574;
+  font-size: 16px;
+  height: 32px;
+  padding-left: 4px;
+
+  &:: placeholder {
+    font-size: 16px;
+  }
+
+  ${({ size }) => SIZE_CSS[size]}
+`;
+
+const TextAreaRoot = styled.textarea<{
+  size: TextInputSize;
+}>`
+  background-color: #fff;
+  border: solid 1px #576574;
+  border-radius: 4px;
+  color: #576574;
+  font-size: 16px;
+  padding-left: 4px;
+
+  &:: placeholder {
+    font-size: 16px;
+  }
+
+  ${({ size }) => SIZE_CSS[size]}
+`;
+
+const SIZE_CSS: Record<TextInputSize, FlattenSimpleInterpolation> = {
+  [TextInputSize.fullWidth]: css`
+    width: 100%;
+  `,
+  [TextInputSize.large]: css`
+    width: 320px;
+  `,
+  [TextInputSize.regular]: css`
+    width: 160px;
+  `,
+  [TextInputSize.small]: css`
+    width: 80px;
+  `
+};
```

Comment:
Probably you didn't cover that there's "large" text input UI in the rough screen image.

<img width="656" alt="Screen Shot 2020-03-16 at 8 12 07 AM" src="https://user-images.githubusercontent.com/4289883/76772271-d9af8680-675d-11ea-8c99-37f31c4f3e85.png">

Reply:
mmm, I covered with fullwidth for that